routers/web/repo/issue_list.go

```diff
@@ -607,7 +642,9 @@ func issues(ctx *context.Context, milestoneID, projectID int64, isPullOption opt
 
 	var issues issues_model.IssueList
 	{
-		ids, err := issueIDsFromSearch(ctx, keyword, &issues_model.IssuesOptions{
+		// Do not repeat the keyword search, since if we had any keyword matches we should
+		// already have their IDs available in keywordMatchedIssueIds.
+		ids, err := issueIDsFromSearch(ctx, "", &issues_model.IssuesOptions{
```

Comment:
Do we really need to make other search engines to support "IssueIDs"?

By reading the code, if "keyword" is empty, the search is always done by database.

![image](https://github.com/user-attachments/assets/124cc0ed-ff30-45c8-b654-70858d596247)

Reply:
Correct.  I only made the others support `IN` to keep the processing of SearchOptions consistent across all of them.

For this actual feature, only the DB indexer will ever be used to do the `IN` (see comment above).

Reply:
So we can drop the "non-database search engine IssueIDs support"? And maybe add some error logs if the IssueIDs would be passed in to avoid abuse?

Reply:
> So we can drop the "non-database search engine IssueIDs support"? And maybe add some error logs if the IssueIDs would be passed in to avoid abuse?

Seems reasonable.

Reply:
https://github.com/go-gitea/gitea/pull/33206/commits/2937c6d6ba4a464885bd2188b0c2272cbb265f7b